src/lib.rs

```diff
@@ -168,6 +152,100 @@ pub fn run(config: &Config) -> Vec<ViolationWithContext> {
     all_violations.into_values().flatten().collect()
 }
 
+#[allow(clippy::too_many_arguments)]
+fn run_match<'a>(
+    query_match: &QueryMatch,
+    aggregated_queries: &'a AggregatedQueries,
+    language: SupportedLanguage,
+    path: &'a Path,
+    file_contents: impl Into<RopeOrSlice<'a>>,
+    config: &'a Config,
+    instantiated_per_file_rules: &mut HashMap<RuleName, Box<dyn RuleInstancePerFile>>,
+    mut on_found_violations: impl FnMut(Vec<ViolationWithContext>),
+    mut on_found_pending_fixes: impl FnMut(Vec<PendingFix>, &InstantiatedRule),
+) {
+    let file_contents = file_contents.into();
+    let (instantiated_rule, rule_listener_index, capture_index_if_per_capture) = aggregated_queries
+        .get_rule_and_listener_index_and_capture_index(language, query_match.pattern_index);
+    match capture_index_if_per_capture {
+        Some(capture_index) => {
+            query_match
+                .captures
+                .into_iter()
+                .filter(|capture| capture.index == capture_index)
+                .for_each(|capture| {
+                    run_single_on_query_match_callback(
+                        path,
+                        file_contents,
+                        instantiated_rule,
+                        instantiated_per_file_rules,
+                        config,
+                        language,
+                        rule_listener_index,
+                        capture.node.into(),
+                        &mut on_found_violations,
```

Comment:
Clippy suggested this, I guess `&mut impl FnMut(...) -> ...` also implements `FnMut(...) -> ...`?

That's good to know, would've made some signatures a little less convoluted in the Typescript compiler for Rust code-base